apps/server-asset-sg/src/features/assets/sync/asset-sync.service.ts

```diff
@@ -0,0 +1,90 @@
+import fs from 'fs/promises';
+import { Injectable, OnApplicationBootstrap } from '@nestjs/common';
+import { SchedulerRegistry } from '@nestjs/schedule';
+import { CronJob } from 'cron';
+import { AssetRepo } from '@/features/assets/asset.repo';
+import { AssetSearchService } from '@/features/assets/search/asset-search.service';
+
+@Injectable()
+export class AssetSyncService implements OnApplicationBootstrap {
+  constructor(
+    private readonly assetSearchService: AssetSearchService,
+    private readonly schedulerRegistry: SchedulerRegistry,
+    private readonly assetRepo: AssetRepo
+  ) {}
+  async onApplicationBootstrap() {
+    const syncFileExists = await this.isSyncRunning();
+    if (syncFileExists) {
+      void fs.rm(assetSyncFile);
+    }
+
+    if (process.env.ANONYMOUS_MODE === 'true') {
+      console.log('Anonymous Mode is activated. Search Index will be automatically synced.');
+      await this.startSyncIfIndexOutOfSync();
+
+      const every20Minutes = '*/20 * * * *';
```

Comment:
Isn't this a bit much? As far as I understood, the data is only updated once a day dunring the night, so there shouldn't be a need to update the sync every 20 minutes, right?